app/commands/exercise/representation/create_or_update.rb

```diff
@@ -0,0 +1,25 @@
+class Exercise
+  class Representation
+    class CreateOrUpdate
+      include Mandate
+
+      initialize_with :submission, :ast, :ast_digest, :mapping, :last_submitted_at
+
+      def call
+        representation = Exercise::Representation.create_or_find_by!(exercise:, ast_digest:) do |rep|
+          rep.source_submission = submission
+          rep.ast = ast
+          rep.mapping = mapping
+          rep.last_submitted_at = last_submitted_at
+        end
+
+        representation.update!(last_submitted_at:) unless representation.new_record?
```

Comment:
It should never be possible for `representation` to be a new record here, as the `create_or_find_by!` has a `!` at the end.